vm/src/stdlib/collections.rs

```diff
@@ -205,10 +205,25 @@ mod _collections {
         }
 
         #[pymethod]
-        fn extendleft(&self, iter: PyIterable, vm: &VirtualMachine) -> PyResult<()> {
-            for elem in iter.iter(vm)? {
-                self.appendleft(elem?);
+        fn extendleft(&self, iter: PyObjectRef, vm: &VirtualMachine) -> PyResult<()> {
+            let max_len = self.maxlen;
+            let mut elements: Vec<PyObjectRef> = vm.extract_elements(&iter)?;
+            elements.reverse();
+
+            if let Some(max_len) = max_len {
+                if max_len > elements.len() {
+                    let mut deque = self.borrow_deque_mut();
+                    let truncate_until = max_len - elements.len();
+                    deque.truncate(truncate_until);
+                } else {
+                    self.borrow_deque_mut().clear();
+                    elements.drain(..(elements.len() - max_len));
```

Comment:
I'm sorry I missed this last time (I thought I had written that both should be changed). This should be `elements.truncate(max_len)` and not a `drain`.

We've reversed the elements and have them in a correct order, if `max_len > elements.len()` we only need to keep the first `max_len` items in `elements` i.e `truncate(max_len)`.